src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Serialization.cs

```diff
@@ -181,26 +181,49 @@ public void WriteTo(ObjectWriter writer)
                     }
                 }
             }
+
+            // sortedNodes is an array of Node instances which is often sorted by Node.Name by the caller. This method
+            // produces a sequence of spans within sortedNodes for Node instances that all have the same Name, allowing
+            // serialization to record the string once followed by the remaining properties for the nodes in the group.
+            static IEnumerable<ReadOnlyMemory<Node>> GroupByName(ImmutableArray<Node> sortedNodes)
+            {
+                if (sortedNodes.IsEmpty)
+                    yield break;
+
+                var startIndex = 0;
+                var currentName = sortedNodes[0].Name;
+                for (var i = 0; i < sortedNodes.Length; i++)
+                {
+                    var node = sortedNodes[i];
+                    if (node.Name != currentName)
+                    {
+                        yield return sortedNodes.AsMemory()[startIndex..i];
+                        startIndex = i;
+                    }
+                }
+
+                yield return sortedNodes.AsMemory()[startIndex..sortedNodes.Length];
+            }
         }
 
         private static SymbolTreeInfo TryReadSymbolTreeInfo(
             ObjectReader reader,
             Checksum checksum,
-            Func<string, ImmutableArray<Node>, Task<SpellChecker>> createSpellCheckerTask)
+            Func<ImmutableArray<Node>, Task<SpellChecker>> createSpellCheckerTask)
         {
             try
             {
-                var concatenatedNames = reader.ReadString();
-
                 var nodeCount = reader.ReadInt32();
                 var nodes = ArrayBuilder<Node>.GetInstance(nodeCount);
-                for (var i = 0; i < nodeCount; i++)
+                while (nodes.Count < nodeCount)
                 {
-                    var start = reader.ReadInt32();
-                    var length = reader.ReadInt32();
-                    var parentIndex = reader.ReadInt32();
-
-                    nodes.Add(new Node(new TextSpan(start, length), parentIndex));
+                    var groupCount = reader.ReadInt32();
+                    var name = reader.ReadString();
+                    for (var i = 0; i < groupCount; i++)
```

Comment:
seeing this def makes me prefer the idea of having this be `name, length, items`

Reply:
➡️ Tracking this under https://github.com/dotnet/roslyn/pull/48896#discussion_r513009479